pkg/s3wrapper/filesystem.go

```diff
@@ -0,0 +1,215 @@
+	log := logutil.FromContext(ctx, f.log)
+	filePath := filepath.Join(f.basedir, objectName)
+	if err := os.MkdirAll(path.Dir(filePath), 0755); err != nil {
+		err = errors.Wrapf(err, "Unable to create directory for file data %s", filePath)
+		log.Error(err)
+		return err
+	}
+	if err := ioutil.WriteFile(filePath, data, 0600); err != nil {
+		err = errors.Wrapf(err, "Unable to write data to file %s", filePath)
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (f *FSClient) UploadFile(ctx context.Context, filePath, objectName string) error {
+	log := logutil.FromContext(ctx, f.log)
+	data, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		err = errors.Wrapf(err, "Unable to open file %s for upload", filePath)
+		log.Error(err)
+		return err
+	}
+	return f.Upload(ctx, data, objectName)
+}
+
+func (f *FSClient) UploadStream(ctx context.Context, reader io.Reader, objectName string) error {
+	log := logutil.FromContext(ctx, f.log)
+	filePath := filepath.Join(f.basedir, objectName)
+	if err := os.MkdirAll(path.Dir(filePath), 0755); err != nil {
+		err = errors.Wrapf(err, "Unable to create directory for file data %s", filePath)
+		log.Error(err)
+		return err
+	}
+	buffer := make([]byte, 1024)
+	for {
+		length, err := reader.Read(buffer)
+		if err != nil {
+			if err != io.EOF {
+				err = errors.Wrapf(err, "Unable to read data for upload to file %s", filePath)
+				log.Error(err)
+				return err
+			}
+			break
+		}
+		f, err := os.OpenFile(filePath, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			err = errors.Wrapf(err, "Unable to open file for writing %s", filePath)
+			log.Error(err)
+			return err
+		}
+		if _, err := f.Write(buffer[0:length]); err != nil {
+			err = errors.Wrapf(err, "Unable to write data to file %s", filePath)
+			log.Error(err)
+			return err
+		}
+		if err := f.Close(); err != nil {
+			err = errors.Wrapf(err, "Unable to close file %s", filePath)
+			log.Error(err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (f *FSClient) Download(ctx context.Context, objectName string) (io.ReadCloser, int64, error) {
+	log := logutil.FromContext(ctx, f.log)
+	filePath := filepath.Join(f.basedir, objectName)
+	fp, err := os.Open(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, 0, NotFound(objectName)
+		}
+		err = errors.Wrapf(err, "Unable to open file %s", filePath)
+		log.Error(err)
+		return nil, 0, err
+	}
+	info, err := fp.Stat()
+	if err != nil {
+		fp.Close()
+		err = errors.Wrapf(err, "Unable to stat file %s", filePath)
+		log.Error(err)
+		return nil, 0, err
+	}
+	return ioutils.NewReadCloserWrapper(fp, fp.Close), info.Size(), nil
+}
+
+func (f *FSClient) DoesObjectExist(ctx context.Context, objectName string) (bool, error) {
+	filePath := filepath.Join(f.basedir, objectName)
+	info, err := os.Stat(filePath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return false, nil
+		}
+		return false, errors.Wrap(err, fmt.Sprintf("failed to get file %s", filePath))
```

Comment:
Wrapf and then you will not need sprtinf

Reply:
Fixed